src/aero_reps/aero_rep_single_particle.c

```diff
@@ -187,6 +187,117 @@ void aero_rep_single_particle_get_effective_radius__m(
+      } else if (LAYER_PHASE_START_(i_layer) <= aero_phase_idx_second &&
+                 aero_phase_idx_second <= LAYER_PHASE_END_(i_layer) &&
+                 i_phase_count == aero_phase_idx_second) {
+        layer_second = i_layer;
+        phase_model_data_id_second = PHASE_MODEL_DATA_ID_(i_layer, i_phase);
+      }
+      ++i_phase_count;
+    }
+  }
+  layer_interface = layer_first > layer_second ? layer_second : layer_first;
+
+  double total_volume_layer_first = 0.0;
+  double total_volume_layer_second = 0.0;
+  double volume_phase_first = 0.0;
+  double volume_phase_second = 0.0;
+  i_phase_count = 0;
+  for (int i_layer = 0; i_layer <= layer_second; ++i_layer) {
+    for (int i_phase = 0; i_phase < NUM_PHASES_(i_layer); ++i_phase) {
+      double *state = (double *)(model_data->grid_cell_state);
+      state += PARTICLE_STATE_SIZE_ + PHASE_STATE_ID_(i_layer,i_phase);
+      double volume;
+      aero_phase_get_volume__m3_m3(model_data, PHASE_MODEL_DATA_ID_(i_layer,i_phase),
+                                   state, &(volume), curr_partial);
+      if (i_layer == layer_first) total_volume_layer_first += volume;
+      if (i_phase_count == aero_phase_idx_first && 
+          PHASE_MODEL_DATA_ID_(i_layer, i_phase) == 
+          phase_model_data_id_first) volume_phase_first = volume;
+      if (i_layer == layer_second) total_volume_layer_second += volume;
+      if (i_phase_count == aero_phase_idx_second && 
+          PHASE_MODEL_DATA_ID_(i_layer, i_phase) == 
+          phase_model_data_id_second) volume_phase_second = volume;
+      ++i_phase_count;
+    }
+  }
+  double f_first = volume_phase_first / total_volume_layer_first;
+  double f_second = volume_phase_second / total_volume_layer_second;
+  double total_volume = 0.0;
+  if (partial_deriv) curr_partial = partial_deriv;
+  for (int i_layer = 0; i_layer <= layer_interface; ++i_layer) {
+    for (int i_phase = 0; i_phase < NUM_PHASES_(i_layer); ++i_phase) {
+      double *state = (double *)(model_data->grid_cell_state);
+      state += PARTICLE_STATE_SIZE_ + PHASE_STATE_ID_(i_layer,i_phase);
+      double volume;
+      aero_phase_get_volume__m3_m3(model_data, PHASE_MODEL_DATA_ID_(i_layer,i_phase),
+                                   state, &(volume), curr_partial);
+      if (partial_deriv) curr_partial += PHASE_NUM_JAC_ELEM_(i_layer,i_phase);
+      total_volume += volume;
+    }
+  }
+  radius = pow(((total_volume) * 3.0 / 4.0 / 3.14159265359), 1.0 / 3.0);
+  *surface_area = f_first * f_second * 4 * 3.14159265359 * pow(radius, 2.0);
+  if (!partial_deriv) return;
+  for (int i_layer = 0; i_layer < NUM_LAYERS_; ++i_layer) {
+    for (int i_phase = 0; i_phase < NUM_PHASES_(i_layer); ++i_phase) {
+      for (int i_spec = 0; i_spec < PHASE_NUM_JAC_ELEM_(i_layer,i_phase); ++i_spec) {
+        if (i_layer <= layer_interface) {
+          *partial_deriv =
+              2.0 * f_first * f_second * pow(radius, -1.0)  * (*partial_deriv);
```

Comment:
This looks like it does not account for `d f_first/d X` and `d f_second/d X` where X are condensed-phase species concentrations.